lib/authutil/authutil.go

```diff
@@ -137,6 +137,18 @@ func getLDAPConnection(u url.URL, timeoutSecs uint, rootCAs *x509.CertPool) (*ld
 	return conn, server, nil
 }
 
+func CheckLDAPConnection(u url.URL, timeoutSecs uint, rootCAs *x509.CertPool) error {
+	conn, _, err := getLDAPConnection(u, timeoutSecs, rootCAs)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	timeout := time.Duration(time.Duration(timeoutSecs) * time.Second)
+	conn.SetTimeout(timeout)
+	conn.Start()
+	return nil
```

Comment:
This always returns OK? Regardless of timeouts?

Reply:
yes